src/VirtualClient/VirtualClient.Actions/HPLinpack/HPLinpackExecutor.cs

```diff
@@ -0,0 +1,408 @@
+            this.stateManager = this.systemManagement.StateManager;
+        }
+
+        /// <summary>
+        /// The order of the coefficient matrix also known as problem size (N)
+        /// </summary>
+        public string ProblemSizeN
+        {
+            get
+            {
+                return this.Parameters.GetValue<string>(nameof(HPLinpackExecutor.ProblemSizeN), Environment.ProcessorCount * 10000);
+            }
+        }
+
+        /// <summary>
+        /// The partitioning blocking factor also known as block size (NB)
+        /// </summary>
+        public string BlockSizeNB
+        {
+            get
+            {
+                this.Parameters.TryGetValue(nameof(HPLinpackExecutor.BlockSizeNB), out IConvertible nb);
+                return nb?.ToString();
+            }
+        }
+
+        /// <summary>
+        /// The name of the package where the ARMPerformanceLibraries package is downloaded.
+        /// </summary>
+        public string ARMPerformanceLibrariesPackageName
+        {
+            get
+            {
+                this.Parameters.TryGetValue(nameof(HPLinpackExecutor.ARMPerformanceLibrariesPackageName), out IConvertible armperformancelibraries);
+                return armperformancelibraries?.ToString();
+            }
+        }
+
+        /// <summary>
+        /// Version of HPL being used.
+        /// </summary>
+        public string Version
+        {
+            get
+            {
+                this.Parameters.TryGetValue(nameof(HPLinpackExecutor.Version), out IConvertible version);
+                return version?.ToString();
+            }
+        }
+
+        /// <summary>
+        /// The path to the HPL directory.
+        /// </summary>
+        protected string HPLDirectory { get; set; }
+
+        /// <summary>
+        /// The number of Process rows(P).
+        /// </summary>
+        protected int ProcessRows { get; set; }
+
+        /// <summary>
+        /// The number of Process columns(Q).
+        /// </summary>
+        protected int ProcessColumns { get; set; }
+
+        /// <summary>
+        /// Initializes the environment for execution of the HPL workload.
+        /// </summary>
+        protected override async Task InitializeAsync(EventContext telemetryContext, CancellationToken cancellationToken)
+        {
+            this.ThrowIfPlatformIsNotSupported();
+
+            if (this.CpuArchitecture == Architecture.Arm64)
+            {
+                DependencyPath armPerformanceLibrariesPackage = await this.packageManager.GetPackageAsync(this.ARMPerformanceLibrariesPackageName, cancellationToken)
+                .ConfigureAwait(false);
+                string armPackageLibrariesPath = armPerformanceLibrariesPackage.Path;
+                await this.systemManagement.MakeFileExecutableAsync(this.PlatformSpecifics.Combine(armPackageLibrariesPath, "arm-performance-libraries_22.1_Ubuntu-20.04.sh"), this.Platform, cancellationToken).ConfigureAwait(false);
```

Comment:
Is this specific to ARM? Could you do a check for the OS, and also include link/doc on why we need to do this